polyfill/intersection-observer.js

```diff
@@ -703,7 +718,9 @@ IntersectionObserver.prototype._hasCrossedThreshold =
  * @private
  */
 IntersectionObserver.prototype._rootIsInDom = function() {
-  return !this.root || containsDeep(document, this.root);
+  var rootDoc =
+    (this.root && (this.root.ownerDocument || this.root)) || document;
+  return !this.root || containsDeep(rootDoc, this.root);
```

Comment:
I think this is not quite right and possibly the old code is more correct. This check now only tests that a root is inside its parent doc. But the parent doc itself could be actually disconnected.

Reply:
Thanks for catching this! Idk why I modified this - it was probably while trying to get the cross-document test working, but ended up handling the case in `_rootContainsTarget`